src/compiler/parser.ts

```diff
@@ -1104,28 +1110,74 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
-        // Create and prime the scanner before parsing the source elements.
-        let scanner = createScanner(languageVersion, /*skipTrivia*/ true, sourceText, scanError);
-        token = nextToken();
+        return { parse };
+
+        function parse(fileName: string, _sourceText: string, languageVersion: ScriptTarget, _syntaxCursor: SyntaxCursor, setParentNodes?: boolean): SourceFile {

```

Comment:
Because I need to assign to "sourceText".  And if I call it "sourceText" then "```sourceText = sourceText``` won't do anything useful :)